[PAC][Driver] Add `-faarch64-jump-table-hardening` flag

kovdan01 opened the pull request:
The flag is placed together with pointer authentication flags since they
serve the same security purpose of protecting against attacks on control
flow. The flag is not ABI-affecting and might be enabled separately if
needed, but it's also intended to be enabled as part of pauth-enabled
environments (e.g. pauthtest).

See also codegen implementation #97666.

kovdan01 committed:
[PAC][clang] Add `-faarch64-jump-table-hardening` driver flag

kovdan01 commented:
* **#113150** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/113150?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#113149** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/113149?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kovdan01 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

asl commented:
This looks reasonable to me. @ahmedbougacha @ahatanak any objections?

kovdan01 commented:
@ahmedbougacha @ahatanak Please let me know if you have any objections on this.

kovdan01 commented:
@ahmedbougacha @ahatanak Please let me know if you have any objections on this.

kovdan01 commented:
@ahmedbougacha @ahatanak Please let me know if you have any objections on this.
@MaskRay Would be glad to see your feedback as well. See also https://github.com/llvm/llvm-project/pull/113152#issuecomment-2486394507

kovdan01 commented:
@ahmedbougacha @ahatanak Please let me know if you have any objections on this.

MaskRay reviewed (approved):
LGTM

I expect some discussion in the description / commit message why AArch64JumpTableHardening is placed together with other `PointerAuth*` options.

kovdan01 commented:
> LGTM
> 
> I expect some discussion in the description / commit message why AArch64JumpTableHardening is placed together with other `PointerAuth*` options.

@MaskRay Thanks for feedback, please let me know if the following description addresses your comment:

> The flag is placed together with pointer authentication flags since they
> serve the same security purpose of protecting against attacks on control
> flow. The flag is not ABI-affecting and might be enabled separately if
> needed, but it's also intended to be enabled as part of pauth-enabled
> environments (e.g. pauthtest).
>
> See also codegen implementation #97666.

kovdan01 commented:
### Merge activity

* **Dec 5, 3:32 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/113149).
